src/std/rpc/ws_client/client.rs

```diff
@@ -167,4 +165,20 @@ impl WsRpcClient {
                 })?;
         Ok(())
     }
+
+    fn direct_rpc_connect(
+        &self,
+        jsonreq: String,
+        result_in: ThreadOut<String>,
+        on_message_fn: OnMessageFn,
+    ) -> WsResult<()> {
+        connect(self.url.as_str(), |out| RpcClient {
+            out,
+            request: jsonreq.clone(),
+            result: result_in.clone(),
+            on_message_fn,
+        })?;
+
+        Ok(())
+    }
```

Comment:
This looks good, I agree with your thinking, as the whole thing is blocking anyhow, this is merely an internal improvement. However, I wonder about one things:

* Would it not make sense to create the channel internally here, and directly return a `WsResult<String>`? Otherwise, the function looks the same from the outside as the `start_rpc_client_thread` even though it has different semantics.